src/gui/widgets/Graph.cpp

```diff
@@ -200,48 +200,28 @@ void Graph::drawLineAt( int _x, int _y, int _lastx )
 
 	float range = minVal - maxVal;
 	float val = ( _y*range/( height()-5 ) ) + maxVal;
-	
-	int sample_begin, sample_end;
-	float lastval;
-	float val_begin, val_end;
-	
-	if (_lastx > _x)
-	{
-		sample_begin = (int)((_x) * xscale);
-		sample_end = (int)ceil((_lastx+1) * xscale);
-		lastval = model() -> m_samples[ (int)( sample_end - 1 ) ];
-		val_begin = val;
-		val_end = lastval;
 
-	}
-	else
-	{
-		sample_begin = (int)(_lastx * xscale);
-		sample_end =  (int)ceil((_x+1) * xscale);
-		lastval = model() -> m_samples[ (int)( sample_begin ) ];
-		val_begin = lastval;
-		val_end = val;
-		
-	}
-	
-	// calculate line drawing variables
-	int linelen = sample_end - sample_begin;
-	if (linelen == 1)
-	{
-		val_begin = val;
-	}
+	const auto sampleBegin = static_cast<int>((_lastx > _x ? _x : _lastx) * xscale);
+	const auto sampleEnd = static_cast<int>(ceil((_lastx > _x ? _lastx : _x) + 1 * xscale));
+
+	const auto lastVal = model()->m_samples[static_cast<int>(_lastx > _x ? sampleEnd - 1 : sampleBegin)];
+	const auto lineLen = sampleEnd - sampleBegin;
+
+	const auto valBegin = _lastx > _x || lineLen == 1 ? val : lastVal;
+	const auto valEnd = _lastx > _x ? lastVal : val;
```

Comment:
We could probably remove a lot of the duplicate `_lastx > _x` stuff with a prior `const bool pastx = _lastx > _x`